scheduler/scheduler_system.go

```diff
@@ -288,6 +289,12 @@ func (s *SystemScheduler) computeJobAllocs() error {
 	destructiveUpdates, inplaceUpdates := inplaceUpdate(s.ctx, s.eval, s.job, s.stack, updates)
 	diff.update = destructiveUpdates
 
+	for _, inplaceUpdate := range inplaceUpdates {
+		if !allocExistsForTaskGroup[inplaceUpdate.TaskGroup.Name] {
+			allocExistsForTaskGroup[inplaceUpdate.TaskGroup.Name] = true
+		}
```

Comment:
Because you're only consuming this map as an existence check (i.e. effective treating it as a set type), you can set it to `true` unconditionally and save yourself both a conditional branch and hashing the key again:

```suggestion
		allocExistsForTaskGroup[inplaceUpdate.TaskGroup.Name]  = true
```